nat-lab/tests/test_mesh_firewall.py

```diff
@@ -781,5 +774,5 @@ async def conntrack_on_stdout(stdout: str) -> None:
                 # kill client and check what is happening in conntrack events
                 # if everything is correct -> conntrack should show LAST_ACK -> TIME_WAIT
                 # if something goes wrong, it will be stuck at LAST_ACK state
-                await last_ack_event.wait()
-                await time_wait_event.wait()
+                assert await conntrack.event_with_tcp_state_exists(TcpState.LAST_ACK)
+                assert await conntrack.get_last_event_tcp_state() == TcpState.TIME_WAIT
```

Comment:
You are introducing a race here. Previously we had a wait here for the given event. Now you just check the list of currently received events. It's possible that those LAST_ACK event has not been yet registered by conntrack. Please add a wait for it otherwise we'll see a flakiness from it

Reply:
Hmm you're right, I was afraid this might be the case.. Internally get events call waits for the synchronize event, since takes a bit of time everything appears to be arriving in the correct order, according to logs. Although theoretically there is no guarantee that sync event will arrive after LAST_ACK event, I'll try to think of a more elegant way to implement this.